File: python/countingsort.py
#!/usr/bin/python

"""Sorts CLI arguments using counting sort"""

import itertools
import sys


def sort(array):
    """Sorts a list of numbers using counting sort"""
    greatest = max(array)
    counts = [0] * (greatest + 1)
    outputs = []
    for i in array:
        counts[i] += 1
    for index, i in enumerate(counts):
        for _ in itertools.repeat(1, i):
            outputs.append(index)
    return outputs


def main():
    """Program entry point"""
    inputs = list(map(int, sys.argv[1:]))
    outputs = sort(inputs)
    for i in outputs:
        print(i, end=' ')
    print('')


if __name__ == '__main__':
    main()

File: rb/countingsort.rb
#!/usr/bin/env ruby
#
def sort(input)
	output = []
	counts = Array.new(input.max + 1, 0)
	input.each do |i|
		counts[i] += 1
	end
	for i in 0..(counts.size - 1)
		counts[i].times do
			output << i
		end
	end
	return output
end

inp = []
while i = STDIN.gets
	inp << i.to_i
end
p sort inp
